### PR TITLE
Asm tokens fixes

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes multiple issue with the asm token implementation:

- Several resource leaks (CID 396039 - 396057, not in the unit tests though).
- Replace `c55x` test with `c55x+` test since `c55x` is broken in Capstone if run on `s390x`
- Replace arm token color test with one which produces the same asm string over Capstone v3-v4
- Revive an unreachable code snippet. 

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/asm/asm.c [hunk 1]
```diff
@@ -1323,7 +1323,10 @@ RZ_API RZ_OWN RzAsmTokenString *rz_asm_token_string_clone(RZ_OWN RZ_NONNULL RzAs
 	new->str = rz_strbuf_new(rz_strbuf_get(toks->str));
 	new->op_type = toks->op_type;
 
-	rz_return_val_if_fail(new->tokens &&new->str, NULL);
+	if (!(new->tokens &&new->str)) {
```
Comment (wargio) [inline librz/asm/asm.c]: not formatted.

Comment (Rot127) [reply librz/asm/asm.c]: The CI rejected the space in `if (!(new->tokens && new->str)) {`.

Comment (wargio) [reply librz/asm/asm.c]: what, that is weird.

Comment (wargio) [reply librz/asm/asm.c]: i know why, rename that `new` pointer to something else. `new` is a keyword and should not be used.

Comment (Rot127) [reply librz/asm/asm.c]: Ah, right. Thanks

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**